src/lib.rs

```diff
@@ -1667,13 +1681,26 @@ fn ns_from_dur(dur: Duration) -> u64 {
     dur.as_secs() * 1_000_000_000 + u64::from(dur.subsec_nanos())
 }
 
+#[inline(never)]
+#[allow(clippy::needless_pass_by_value)]
+fn black_box<T>(x: T) -> T {
+    #[cfg(all(feature = "unstable", not(stage0)))]
+    {
+        std::hint::black_box(x)
+    }
+    #[cfg(any(not(feature = "unstable"), stage0))]
+    {
+        unsafe { std::ptr::read_volatile(&x as *const T) }
```

Comment:
Mmm, doesn't `x`'s destructor execute immediately after this line? I think a mem::forget or something is needed after the `read_volatile`?

Reply:
Indeed, care submitting a PR? Otherwise, I'll fix that.

Reply:
Can't submit PR to PR, i think... You can fix that yourself.

Reply:
You can submit a PR to the clippy_ci branch on gnzlbg's fork

Reply:
https://github.com/gnzlbg/libtest/pull/1